# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/a-demesa/flyrank-ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the **Refresh / Content Opportunity Scoring** lane because it focuses on identifying pages that are good candidates for content updates. Instead of refreshing every page, this project aims to prioritize pages that are more likely to benefit from an update based on observable signals in the data. This can help save time and support better decisions about which content should be reviewed first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision**

Which pages should be prioritized for content refresh?

**Action**

An SEO specialist or content manager can update the highest-priority pages first.

**Cost of a wrong recommendation**

If pages are incorrectly prioritized, valuable pages may continue losing traffic while time is spent updating pages that provide little benefit.

**Why data or ML helps**

Machine learning can analyze multiple signals together instead of relying on simple rules, making the prioritization process more informed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import os
import subprocess

if not os.path.isdir("flyrank-ml"):
    subprocess.run(["git", "clone", "https://github.com/a-demesa/flyrank-ml.git"], check=True)

os.chdir("flyrank-ml")

print(os.getcwd())

/content/flyrank-ml/flyrank-ml/flyrank-ml/flyrank-ml


In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Pages with downward trend:", (df["trend_direction"] == "down").sum())
print("Average CTR:", round(df["ctr"].mean(), 3))

Total pages: 30000
Pages with downward trend: 16262
Average CTR: 0.511


The starter dataset contains **30,000 pages**, which provides enough data for meaningful analysis.

Out of these, **16,262 pages** have a downward trend, indicating that many pages could potentially benefit from a content refresh.

The average click-through rate (CTR) is **0.511**, showing that page performance varies and that multiple signals can be used to prioritize which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify observed patterns in the available data and provide decision support for prioritizing content refresh opportunities.

The results should be interpreted as observed relationships rather than proof of cause and effect. This project does not predict Google's ranking algorithm or guarantee that refreshing a page will improve its performance.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.